File: app/src/main/java/ua/hospes/nfs/marathon/core/di/components/AppComponent.java
package ua.hospes.nfs.marathon.core.di.components;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import ua.hospes.nfs.marathon.App;
import ua.hospes.nfs.marathon.core.di.module.ActivitiesModule;
import ua.hospes.nfs.marathon.core.di.module.AppModule;

/**
 * @author Andrew Khloponin
 */
@Singleton
@Component(modules = {AndroidInjectionModule.class, AppModule.class, ActivitiesModule.class})
public interface AppComponent {
    void inject(App app);
}

File: app/src/main/java/ua/hospes/rtm/di/AppComponent.kt
package ua.hospes.rtm.di

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import ua.hospes.rtm.AppBase
import ua.hospes.rtm.di.module.*
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    AppModule::class, DbModule::class, NetModule::class, RepoModule::class,
    ActivitiesModule::class])
interface AppComponent : AndroidInjector<AppBase> {
    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<AppBase>()
}